include/irr/asset/IAssetLoader.h

```diff
@@ -7,9 +7,47 @@
 namespace irr { namespace asset
 {
 
+//! A class that defines rules during Asset-loading process
+/**
+	Every Asset must be loaded by a particular class derived from IAssetLoader.
+	These classes must be registered with IAssetManager::addAssetLoader() which will 
+	add it to the list of loaders (grab return 0-based index) or just not register 
+	the loader upon failure (don�t grab and return 0xdeadbeefu).
+
+	The loading is impacted by caching and resource duplication flags, defined as IAssetLoader::E_CACHING_FLAGS.
+
+	When the class derived from IAssetLoader is added, its put once on an 
+	std::vector<IAssetLoader*> and once on an std::multimap<std::string,IAssetLoader*> 
+	inside the IAssetManager for every associated file extension it reports.
+
+	The loaders are tried in the order they were registered per file extensions, 
+	and later in the global order in case of needing to fallback to examining files.
+
+	An IAssetLoader can only be removed/deregistered by its original pointer or global loader index.
+
+	@see IReferenceCounted::grab()
+	@see IAsset
+	@see IAssetManager
+	@see IAssetWriter
+*/
+
 class IAssetLoader : public virtual core::IReferenceCounted
 {
 public:
+
+	//! Caching and resource duplication flags 
```

Comment:
its a rather weird bitfield, we'd have to explain how the loading process works.

Basically a mesh can reference a submesh, a submesh can reference a material, a material can reference a texture, etc.

If you look at them as recursive functions, the `LEVEL` is basically how deep down the stack you are.
This flag is a bitfield with 2 bits per level, and the enums provie are just some useful constants. But different combinations are valid as well.

Example
```cpp
static_cast<E_CACHING_FLAGS>(ECF_DONT_CACHE_TOP_LEVEL>>4ull) // anything on level 2 (top is 0) will not get cached
```